packages/dynamite/dynamite/lib/src/models/discriminator.dart

```diff
@@ -1,30 +1,16 @@
-import 'package:collection/collection.dart';
-import 'package:json_annotation/json_annotation.dart';
-import 'package:meta/meta.dart';
+import 'package:built_value/built_value.dart';
+import 'package:built_value/serializer.dart';
 
 part 'discriminator.g.dart';
 
-@JsonSerializable()
-@immutable
-class Discriminator {
-  const Discriminator({
-    required this.propertyName,
-    this.mapping,
-  });
+abstract class Discriminator implements Built<Discriminator, DiscriminatorBuilder> {
+  factory Discriminator([final void Function(DiscriminatorBuilder) updates]) = _$Discriminator;
 
-  factory Discriminator.fromJson(final Map<String, dynamic> json) => _$DiscriminatorFromJson(json);
-  Map<String, dynamic> toJson() => _$DiscriminatorToJson(this);
+  Discriminator._();
```

Comment:
I usually like making them const :woman_shrugging: 

This is nothing functional and rather a preference of code style